portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,25 +38,33 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    PreparedQuery results = datastore.prepare(new Query("Comment"));
-    List<String> comments = new ArrayList<>();
+    PreparedQuery results =
+        datastore.prepare(new Query("Comment").addSort("timestamp", SortDirection.DESCENDING));
+    List<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
       String message = (String) entity.getProperty("message");
-      comments.add(message);
+      long timestamp = (long) entity.getProperty("timestamp");
+      comments.add(new Comment(id, name, timestamp, message));
```

Comment:
You can instead create a method like buildComment(Entity);

Reply:
done